x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -326,6 +337,35 @@ public ValueFetcher valueFetcher(SearchExecutionContext context, String format)
         public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext) {
             throw new IllegalArgumentException("[semantic_text] fields do not support sorting, scripting or aggregating");
         }
+
+        public QueryBuilder semanticQuery(InferenceResults inferenceResults, float boost, String queryName) {
+            String nestedFieldPath = getChunksFieldName(name());
+            String inferenceResultsFieldName = getEmbeddingsFieldName(name());
+            QueryBuilder childQueryBuilder;
+
+            if (modelSettings == null) {
+                // No inference results have been indexed yet
+                childQueryBuilder = new MatchNoneQueryBuilder();
+            } else if (inferenceResults instanceof TextExpansionResults textExpansionResults) {
```

Comment:
We might want to check the modelSettings correspond to the inferenceResults. It could be possible that the inferenceId is removed and then re-created with a different task type. 

We want to avoid the user doing that, but should we double check to provide a helpful error message in the meantime?

Reply:
Should we return a 400 or 500 HTTP response in this case? I'm leaning towards 500 since the problem isn't with the request, but with the inference service.

Reply:
I'd say it's a 400, as the inference was working correctly but the inference id is wrong. Seems like a user failure to me as everything else worked.